src/link/seriallink.cpp

```diff
@@ -99,12 +109,146 @@ void SerialLink::setBaudRate(int baudRate)
+        // Configure timeout
+        COMMTIMEOUTS currentCommTimeouts;
+        ::ZeroMemory(&currentCommTimeouts, sizeof(currentCommTimeouts));
+
+        if(::GetCommTimeouts(handle, &currentCommTimeouts)) {
+            currentCommTimeouts.ReadIntervalTimeout = 1;
+            if (!::SetCommTimeouts(handle, &currentCommTimeouts)) {
+                qCWarning(PING_PROTOCOL_SERIALLINK) << "Failed to set read interval timeout.";
+                qCDebug(PING_PROTOCOL_SERIALLINK) << "System error:" << ::GetLastError();
+            }
+        } else {
+            qCWarning(PING_PROTOCOL_SERIALLINK) << "Failed to get COMMTIMEOUTS struct from system.";
+            qCDebug(PING_PROTOCOL_SERIALLINK) << "System error:" << ::GetLastError();
+        }
+    } else {
+        qCWarning(PING_PROTOCOL_SERIALLINK) << "Failed to get serial handle from OS.";
+    }
+
+    // Set low latency mode with windows register system
+    // HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\FTDIBUS\VID_XXXX+PID_XXXX+ABCDEFGH\0000\Device Parameters\LatencyTimer
+
+    // We need to close connection before changing the register
+    _port.close();
+
+    // Create VID/PID/serial number register string path
+    QSerialPortInfo serialPortInfo(_port);
+    const QString deviceSettingsIdentification = QStringLiteral("VID_%1+PID_%2+%3")
+            .arg(serialPortInfo.vendorIdentifier(), 4, 16, QChar('0'))
+            .arg(serialPortInfo.productIdentifier(), 4, 16, QChar('0'))
+            .arg(serialPortInfo.serialNumber());
+    const QString settingsSystemKey =
+        QStringLiteral(
+            R"(HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\FTDIBUS\%1\0000\Device Parameters\)")
+        .arg(deviceSettingsIdentification);
+    qCDebug(PING_PROTOCOL_SERIALLINK) << "Register path:" << settingsSystemKey;
+
+    // Change register LatencyTimer value to 1ms
+    QSettings settings(settingsSystemKey, QSettings::NativeFormat);
+
+    // Variable exist, so our path is valid
+    QVariant latencyTimerValue = settings.value("LatencyTimer");
+    if(!latencyTimerValue.isValid()) {
+        qCWarning(PING_PROTOCOL_SERIALLINK) << "Variable does not exist, register path is wrong.";
+        return false;
+    }
+
+    if(latencyTimerValue.toInt() == 1) {
+        qCDebug(PING_PROTOCOL_SERIALLINK) << "Low latency timer is already in register.";
+        return true;
+    }
+
+    if(!settings.isWritable()) {
+        qCWarning(PING_PROTOCOL_SERIALLINK) << "Register is not writable.";
+        return false;
+    }
+
+    settings.setValue("LatencyTimer", 1);
+    settings.sync();
+    auto settingsStatus = settings.status();
+    if(settingsStatus != QSettings::NoError) {
+        qCWarning(PING_PROTOCOL_SERIALLINK) << "Failed to save LatencyTimer setting.";
+        qCDebug(PING_PROTOCOL_SERIALLINK) << "Settings status:" << settingsStatus;
+        return false;
+    }
+    startConnection();
+#endif
+
+    //light at the end of the tunnel
+    return true;
+
+#endif
+}
+
```

Comment:
break that function into two, one that handles windows, another one that handles linux? it's hard to read with the ifdef blogs getting the whole function.

Reply:
I would like to avoid such thing, because this will need ifdefs in header and ifdefs in cpp. Let me know if you a better solution.